### PR TITLE
Add com.hamrick.VueScan

### PR DESCRIPTION
**Pull request by FakeShemp**
VueScan is a scanning program that is compatible with over 5680 scanners - and is available for Windows, Mac OS X and Linux.

https://www.hamrick.com/

Seems to work fine as far as I can tell.

### CODE REVIEW

**Comment by AsciiWolf** [comment]
bot, build com.hamrick.VueScan

**Comment by FakeShemp** [comment]
Seems to build
https://flathub.org/builds/#/builders/2/builds/3619

File: com.hamrick.VueScan.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (eszlari) [inline com.hamrick.VueScan.appdata.xml]: You have to add at least one screenshot.

File: com.hamrick.VueScan.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<component type="desktop">
+	<id>com.hamrick.VueScan.desktop</id>
```
Comment (eszlari) [inline com.hamrick.VueScan.appdata.xml]: ".desktop" can be dropped

File: com.hamrick.VueScan.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+app-id: com.hamrick.VueScan
+runtime: org.gnome.Platform
+runtime-version: 3.28
+sdk: org.gnome.Sdk
+command: vuescan
+
+tags:
+- proprietary
```
Comment (eszlari) [inline com.hamrick.VueScan.yaml]: indentation: https://github.com/flathub/flathub/wiki/YAML-Style-Guide

File: com.hamrick.VueScan.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+- --persist=.vuescan
+- --device=all
+
+modules:
+- name: VueScan
+  buildsystem: simple
+  build-commands:
+  - install -D apply_extra /app/bin/apply_extra
+  - install -D vuescan /app/bin/vuescan
+  - install -Dm644 com.hamrick.VueScan.appdata.xml /app/share/appdata/com.hamrick.VueScan.appdata.xml
```
Comment (eszlari) [inline com.hamrick.VueScan.yaml]: If you use "-t" you don't have to repeat the file name.

File: com.hamrick.VueScan.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    - rm vuescan.tgz
+    - mv VueScan/* .
+    # Unfortunately the SVG is just a JPEG embedded in SVG container
+    - rm vuescan.svg
+    - rmdir VueScan
+    
+  - type: script
+    dest-filename: vuescan
+    commands:
+    - 'exec /app/extra/vuescan "$@"'
```
Comment (eszlari) [inline com.hamrick.VueScan.yaml]: Quotes can be dropped.

File: com.hamrick.VueScan.yaml [hunk 6]
```diff
@@ -0,0 +1,79 @@
+app-id: com.hamrick.VueScan
+runtime: org.gnome.Platform
+runtime-version: 3.28
```
Comment (eszlari) [inline com.hamrick.VueScan.yaml]: this should be a string -> quotes

File: com.hamrick.VueScan.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+	<name>VueScan</name>
+	<project_license>LicenseRef-proprietary</project_license>
+	<developer_name>Hamrick Software</developer_name>
+	<summary>Whether you need to scan a single page or a hundred page book, VueScan has the tools to help you.</summary>
```
Comment (eszlari) [inline com.hamrick.VueScan.appdata.xml]: too long, see https://www.flathub.org

**Comment by nedrichards** [comment]
That's.... not ideal. We have some checker infrastructure at endless but build whack-a-mole isn't fun for anyone. How old is 9.5.x?

**Comment by nedrichards** [comment]
bot, build com.hamrick.VueScan

**Comment by FakeShemp** [comment]
It's from December 2017.

**Comment by nedrichards** [comment]
That doesn't sound too ridiculous, and is a smart workaround.

**Comment by barthalion** [comment]
As usually, I question if we want another closed-source extra-data app without upstream involvement that isn't really popular or a game.

**Comment by nedrichards** [comment]
You need to add the:
> --share=network 
permission otherwise network scanners (like mine) can't be used. I tested with it enabled on the command line and it worked well.

@barthalion I'm sympathetic to your point, but disagree with it. This is a high quality, stable app that actually works out of the box. It's a good fit for flathub. It could be a *better* fit, of course, but having a scanning app available that actually works without arcane twiddling in the base os is kind of useful too.

File: com.hamrick.VueScan.appdata.xml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+<component type="desktop">
+	<id>com.hamrick.VueScan</id>
+	<name>VueScan</name>
+	<project_license>LicenseRef-proprietary</project_license>
+	<developer_name>Hamrick Software</developer_name>
+	<summary>Scanning software for 5000+ scanners</summary>
+	<metadata_license>CC0-1.0</metadata_license>
+	
+	<description>
+		<p>VueScan is a scanning program that is compatible with over 5680 scanners - and is available for Windows, Mac OS X and Linux.</p>
```
Comment (nedrichards) [inline com.hamrick.VueScan.appdata.xml]: I'd like it to be clearer in the description that the version you install out of the box is a free trial and needs payment to unlock.

File: com.hamrick.VueScan.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+	</description>
+	
+	<screenshots>
+		<screenshot type="default">
+			<image type="source">https://raw.githubusercontent.com/FakeShemp/flathub/com.hamrick.VueScan/screenshot.png</image>
```
Comment (nedrichards) [inline com.hamrick.VueScan.appdata.xml]: When this is accepted you should change the URL here to the actual flathub repo.

Comment (FakeShemp) [reply com.hamrick.VueScan.appdata.xml]: For sure :)

**Comment by nedrichards** [comment]
bot, build com.hamrick.VueScan

**Comment by nedrichards** [comment]
So yeah - either switch back to the 9.6 branch or bundle libusb as well (like this https://github.com/flathub/org.DolphinEmu.dolphin-emu/blob/2b73a580ebccebce9a9b2cbf3ac5755af7e11e94/org.DolphinEmu.dolphin-emu.json#L25 for example)

**Comment by nedrichards** [comment]
bot, build com.hamrick.VueScan

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.hamrick.VueScan

**Pull request closed without merging**